### PR TITLE
Only support User with Login users

### PR DESCRIPTION
**Pull request by kburtram**
This is to avoid scenario in bug https://github.com/microsoft/azuredatastudio/issues/22280 for March early preview.

### CODE REVIEW

File: src/Microsoft.SqlTools.ServiceLayer/Security/UserData.cs [hunk 1]
```diff
@@ -1044,46 +1044,47 @@ public UserPrototype GetUserPrototype(ExhaustiveUserTypes userType)
...
+                //     currentData.userType = UserType.SqlUser;
+                //     this.windowsUser ??= new UserPrototypeForWindowsUser(this.context, this.currentData, this.originalData);
+                //     this.currentPrototype = windowsUser;
+                //     break;
                 
                 default:
                     System.Diagnostics.Debug.Assert(false, "Unknown UserType provided.");
                     this.currentPrototype = null;
-                    break;
+                    throw new ApplicationException("Only 'User with Login' user type support in March Preview release");
```
Comment (alanrenmsft) [inline src/Microsoft.SqlTools.ServiceLayer/Security/UserData.cs:1086]: probably we don't have to call out the release?

**Review by alanrenmsft — APPROVED** [review]

File: src/Microsoft.SqlTools.ServiceLayer/Security/UserData.cs (hunk 1 above)
Comment (kburtram) [reply src/Microsoft.SqlTools.ServiceLayer/Security/UserData.cs:1086]: I'm assuming this won't be the case in the next release :)

**Comment by kburtram** [comment]
The previous implementation wasn't working since those checks were too late (after the type was already hardcoded).  This is an alternate implementation that failure earlier.  Below is the UX.

image

**Comment by alanrenmsft** [comment]
LGTM, if you sync to latest main, the dialog should be automatically closed.

**Pull request merged**